app/src/main/java/org/eyeseetea/malariacare/sdk/SdkPullController.java

```diff
@@ -261,7 +261,9 @@ private static void pullOrganisationUnits() {
                     public void call(List<OrganisationUnit> organisationUnits) {
                         Log.d(TAG, "OrganisationUnit: Done");
                         asyncDownloads--;
-                        if (pullData) {
+                        mProgramAndOrganisationUnitWrapper = new ProgramAndOrganisationUnitWrapper();
+                        mProgramAndOrganisationUnitWrapper.buildProgramAndOrganisationUnit(organisationUnits, sdkPrograms);
+                        if(pullData) {
                             loadDataValues();
                         }
                         convertData();
```

Comment:
@idelcano I have a doubt. Does it make sense to invoke ConvertData from here when is invoked from pullEventsByProgramAndOrganisationUnit?